File: backend/fix_models.js
const fs = require('fs');
const content = fs.readFileSync('app/db/models.py', 'utf-8');

const newContent = content.replace(/( +)(__tablename__ = ["']\w+["'])/g, (match, prefix, tablename) => {
    // skip if next line is already extend_existing
    const nextText = content.substring(content.indexOf(match) + match.length, content.indexOf(match) + match.length + 50);
    if (nextText.includes('extend_existing')) {
        return match;
    }
    return `${prefix}${tablename}\n${prefix}__table_args__ = {'extend_existing': True}`;
});

fs.writeFileSync('app/db/models.py', newContent, 'utf-8');
console.log('Fixed models.py with extend_existing=True for all tables.');

File: backend/fix_models.py
import re

with open('app/db/models.py', 'r', encoding='utf-8') as f:
    content = f.read()

def replacer(match):
    prefix = match.group(1)
    tablename_line = match.group(2)
    
    # Don't add if already there
    if '__table_args__' in content[match.end():match.end()+50]:
        return match.group(0)
        
    return f'{prefix}{tablename_line}\n{prefix}__table_args__ = {{"extend_existing": True}}'

new_content = re.sub(r'(^ +)(__tablename__ = [\"\'][\w_]+[\"\'])', replacer, content, flags=re.MULTILINE)

with open('app/db/models.py', 'w', encoding='utf-8') as f:
    f.write(new_content)

print("Done updating models.py")
